pkg/mimirpb/compat.go

```diff
@@ -299,6 +310,67 @@ func fromSpansToSpansProto(s []histogram.Span) []BucketSpan {
 	return spans
 }
 
+// FromPointsToSamples converts []promql.Point to []Sample.
+func FromPointsToSamples(points []promql.Point) []Sample {
+	samples := make([]Sample, 0, len(points))
+	for _, point := range points {
+		if point.H != nil {
+			continue
+		}
+		samples = append(samples, Sample{
+			TimestampMs: point.T,
+			Value:       point.V,
+		})
+	}
+	return samples
+}
+
+// FromPointsToHistograms converts []promql.Point to []FloatHistogramPair.
+func FromPointsToHistograms(points []promql.Point) []FloatHistogramPair {
+	samples := make([]FloatHistogramPair, 0, len(points))
+	for _, point := range points {
+		h := point.H
+		if h == nil {
+			continue
+		}
+		samples = append(samples, FloatHistogramPair{
+			TimestampMs: point.T,
+			Histogram:   *FloatHistogramFromPrometheusModel(point.H),
+		})
+	}
+	return samples
+}
+
+// FromFloatHistogramToPromHistogram converts histogram.FloatHistogram to model.SampleHistogram.
+func FromFloatHistogramToPromHistogram(h *histogram.FloatHistogram) *model.SampleHistogram {
```

Comment:
I don't think we're testing this directly. Although it's used in generating a frontend reply, so integration tests will cover it. We should add a small smoke test though.